src/test/groovy/graphql/TypeResolverExecutionTest.groovy

```diff
@@ -580,6 +582,82 @@ class TypeResolverExecutionTest extends Specification {
                 "__typename": "NewBar",
                 "name"      : "NAME",
         ]]
+    }
+
+    def "can get access to attributes in the type resolver"() {
+        def sdl = '''
+
+        type Query {
+            foo : BarInterface
+        }
+        
+        interface BarInterface {
+             name : String
+        }
+         
+        type NewBar implements BarInterface {
+            name : String
+            newBarOnlyField : String
+        }
 
+        type OldBar implements BarInterface {
+            name : String
+            oldBarOnlyField : String
+        }
+        '''
+
+
+        TypeResolver typeResolver = new TypeResolver() {
+            @Override
+            GraphQLObjectType getType(TypeResolutionEnvironment env) {
+                assert env.getField().getName() == "foo"
+                assert env.getFieldType() == env.getSchema().getType("BarInterface")
+                assert env.getContext() == "Context"
+                assert env.getGraphQLContext().get("x") == "graphqlContext"
+                assert env.getLocalContext() == "LocalContext"
+                return env.getSchema().getObjectType("NewBar")
```

Comment:
we never had a full integration test that ensured that these values get set at runtime